metals/src/main/scala/scala/meta/internal/metals/CodeLensProvider.scala

```diff
@@ -17,134 +12,36 @@ trait CodeLensProvider {
 }
 
 final class DebugCodeLensProvider(
-    buildTargetClasses: BuildTargetClasses,
-    buffers: Buffers,
-    buildTargets: BuildTargets,
-    compilations: Compilations,
+    codeLensProviders: List[CodeLenses],
     semanticdbs: Semanticdbs
-)(implicit ec: ExecutionContext)
-    extends CodeLensProvider {
+) extends CodeLensProvider {
   // code lenses will be refreshed after compilation or when workspace gets indexed
   def findLenses(path: AbsolutePath): Seq[l.CodeLens] = {
-    val lenses = buildTargets
-      .inverseSources(path)
-      .map { buildTarget =>
-        val classes = buildTargetClasses.classesOf(buildTarget)
-        codeLenses(path, buildTarget, classes)
+    semanticdbs
+      .textDocument(path)
+      .documentIncludingStale
+      .map { textDocument =>
+        val doc = TextDocumentWithPath(textDocument, path)
+        codeLensProviders.filter(_.isEnabled).flatMap(_.codeLenses(doc))
       }
-
-    lenses.getOrElse(Nil)
-  }
-
-  private def codeLenses(
-      path: AbsolutePath,
-      target: b.BuildTargetIdentifier,
-      classes: BuildTargetClasses.Classes
-  ): Seq[l.CodeLens] = {
-    semanticdbs.textDocument(path).documentIncludingStale match {
-      case _ if classes.isEmpty =>
-        Nil
-      case Some(textDocument) =>
-        val distance =
-          TokenEditDistance.fromBuffer(path, textDocument.text, buffers)
-
-        for {
-          occurrence <- textDocument.occurrences
-          if occurrence.role.isDefinition
-          symbol = occurrence.symbol
-          commands = {
-            val main = classes.mainClasses
-              .get(symbol)
-              .map(mainCommand(target, _))
-              .getOrElse(Nil)
-            val tests = classes.testClasses
-              .get(symbol)
-              .map(testCommand(target, _))
-              .getOrElse(Nil)
-            main ++ tests
-          }
-          if commands.nonEmpty
-          range <- occurrence.range
-            .flatMap(r => distance.toRevised(r.toLSP))
-            .toList
-          command <- commands
-        } yield new l.CodeLens(range, command, null)
-      case _ =>
-        Nil
-    }
+      .getOrElse(Seq.empty)
   }
 }
 
 object CodeLensProvider {
-  import scala.meta.internal.metals.JsonParser._
-
   private val Empty: CodeLensProvider = (_: AbsolutePath) => Nil
 
   def apply(
-      buildTargetClasses: BuildTargetClasses,
-      buffers: Buffers,
-      buildTargets: BuildTargets,
-      compilations: Compilations,
+      codeLensProviders: List[CodeLenses],
       semanticdbs: Semanticdbs,
       capabilities: ClientExperimentalCapabilities
   )(implicit ec: ExecutionContext): CodeLensProvider = {
     if (!capabilities.debuggingProvider) Empty
```

Comment:
Regarding my comment, I think this is where a change would need to happen. We aren't providing any `codeLensProviders` if the `debuggingProvider` is false, but really the superMethod code lens shouldn't be dependent on this.

Reply:
That should be checked inside the RunTest code lense provider for sure. Thanks for finding it @ckipp01 !

Reply:
ahh okay now I get what this `debuggingProvider` was for!

I moved it to where I create a list of lens providers. 
If debugging is off I only create `superMethodProvider` but if debugging is on I create both.

```scala
    val runTestLensesProvider: List[CodeLenses] = if (clientExperimentalCapabilities.debuggingProvider) {
      List(new RunTestLensesProvider(buildTargetClasses, buffers, buildTargets))
    } else {
      List.empty
    }
```

Reply:
Pushed a change, could you double check it is okay?
https://github.com/scalameta/metals/pull/1487/files#diff-f8c05eebbf12c9c21a7d568f09b500eaR395-R402

Reply:
Woo hoo! Works now 👍 

![2020-03-18 12 34 13](https://user-images.githubusercontent.com/13974112/76984147-d3afd600-6914-11ea-8668-51e070e8f4e8.gif)